README.md

```diff
@@ -1 +1,40 @@
-# git
\ No newline at end of file
+# git
+
+## Intro
+* Git : A tool track changes in your code 
+* Github : Website that hosts the Git repositories 
+___
+## git terms 
+* clone : make local copy of the remote repo
+* add : tracks the file and changes in git
+* commit : saves file in git
+* push : upload commit to the remote repo
+* pull :  download the changes of remote repo into local repo
+* status : shows the status of the file create, updated or deleted
+* readme.md file : contains information about the repo
+* git init : initializes git repo for the corresponding folder
+* git branch : tells which branch is in use currently  
+* gitbash : used for installing git in Windows OS
+* ssh keys : connects local machine to GitHub account and authenticate private key and public key
+* git merge : merges the file/code with the stated branch
+* pull request (PR) : request to pull your code into another branch, people can review it, make changes
+* stash : temporarily holding the changes, but not committing into master
+* merge conflict : If Developer A tries to edit code that Developer B is editing a conflict may occur.
+* reset : undo the commit changes 
+* fork : propose changes to someone else's project by having a copy in our own account
+___
+## git branches 
+* Main Branch : main branch
+* Feature Branch : replica of main branch, where we make commits 
+* Hot Fix Branch : after or during the merge, if there is code break/error this branch is used to fix them\
+<img src="https://user-images.githubusercontent.com/29035091/212301218-e1214690-2f6e-4539-beef-25f4975f3f11.png" width="800" height="500">
+
+___
+## git workflow
+<img src="https://user-images.githubusercontent.com/29035091/212301108-cbb379a9-1a7b-42d4-85f5-885c10d5cea0.png" width="800" height="1000">
```

Comment:
I want you to understand every step in git workflow.

Reply:
Sure, will go through it